rwengine/src/script/modules/GameModule.cpp

```diff
@@ -701,6 +767,31 @@ void game_set_zone_ped_group(const ScriptArguments& args)
 	}
 }
 
+bool game_has_respray_happened(const ScriptArguments& args)
+{
+	const auto& garages = args.getWorld()->state->garages;
+	int garageIndex = args[0].integerValue();
+
+printf("Checking garage %d for respray!\n", garageIndex);
+	RW_CHECK(garageIndex >= 0, "Garage index too small");
+	RW_CHECK(garageIndex < static_cast<int>(garages.size()), "Garage index too large");
+  const auto& garage = garages[garageIndex];
+
+  if (garage.type != GarageInfo::GARAGE_RESPRAY) {
+    return false;
+  }
+
+	auto playerobj = args.getWorld()->pedestrianPool.find(args.getState()->playerObject);
```

Comment:
/// @todo should be moved to garage code and wait for actual respray to happen (current code only checks if car is in garage)